src/utilities/utilities.ts

```diff
@@ -188,6 +188,29 @@ export function swiftpmSDKFlags(): string[] {
     return [];
 }
 
+/**
+ * Get build path flags to be passed to swift package manager and sourcekit-lsp server
+ */
+export function buildPathFlags(): string[] {
+    if (configuration.buildPath && configuration.buildPath.length > 0) {
+        return ["--build-path", configuration.buildPath];
+    } else {
+        return [];
+    }
+}
+
+/**
+ * Get build path from configuration if exists or return a fallback .build directory in given workspace
+ * @param filesystem path to workspace that will be used as a fallback loacation with .build directory
+ */
+export function buildDirectoryFromWorkspacePath(workspacePath: string): string {
+    if (configuration.buildPath && configuration.buildPath.length > 0) {
+        return configuration.buildPath;
```

Comment:
Is it worthwhile supporting relative paths as well?

Reply:
The file is created as a result of running tests. I figures if we are saving test assets anyway it might have sense to keep it in repository. I can remove it if you feel like it should not be in the repository

Reply:
I'm not sure if i correctly understand your question . Are you asking if user can supply ".customBuildFolder" in this setting? It should work as is. The only thing I'm not sure is how vscode.Uri would resolve it. I can test it a bit later and if it's weird - we can just do logic that merger workspace path with relative path in the setting

Reply:
Yeah I guess you're right with regards to parameters for swift calls, the relative path will work. launch.json configs will probably need prefixed with `${workspaceFolder:<folder name>}`

Reply:
I tested relative build paths and the `launch.json` configs look to work fine with a relative path. I have noted two cases where it does not work though. When loading `workspace-state.json` and when searching for files in package dependency view.

Would it be possible to add a second parameter `fullpath` to `buildDirectoryFromWorkspacePath` which indicates we want a non relative path. If it is set to true and `buildPath` doesn't start with a '/' then return `path.join(workspacePath, configuration.buildPath);`. In the two cases I mentioned above where it needs the full path, we should set this flag to true.

I don't want a full path all the time because I would rather not pollute the `launch.json` with details of where the project is installed.

Reply:
Checking for "/" prefix is easy. what's not easy is to do the same for windows disks. like, needs a regex and, to be honest, i don't remember windows disk naming rules anymore. is it always single character? virtual disks can have more than one? can disk label contain numbers?

Reply:
damn windows. there are also network disks and paths to files on those can start with "\\"

Reply:
the good news is the fact that node's module "path" is smart enough to make this work